src/components/sidebar/right-sidebar/Filters.tsx

```diff
@@ -50,14 +52,41 @@ export default function Filters({filters}: FilterProps) {
     return formatted_options;
   }
 
-  const addFilter = (data: any, filter_type: string) => {
+  const addFilter = async (data: any, filter_type: string) => {
     dispatch({
       type: 'UPDATE_FILTER',
       payload: {
         filter_type,
         filter_value: data.value
       }
+    });   
+  
+    const api = new httpClient()
+    // Update current records when called
+    const records = await api.getAirtableRecords(state.filters)
+    dispatch({
+      type: 'GET_AIRTABLE_RECORDS',
+      payload:  { records }
     });
+
+    if(state.dataPageState.exploreIsOpen) {        

```

Comment:
better way to do this:

dispatch({
        type: 'SAVE_PAGE_STATE',
        payload: state.dataPageState.exploreIsOpen
})

Reply:
love it